cinema/views.py

```diff
@@ -34,25 +37,84 @@
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    @staticmethod
+    def _params_to_ints(query_string):
+        return [int(str_id) for str_id in query_string.split(",")]
+
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
 
         if self.action == "retrieve":
             return MovieDetailSerializer
 
-        return MovieSerializer
+        return self.serializer_class
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+        date = self.request.query_params.get("date")
+        if actors:
+            actors = self._params_to_ints(actors)
+            queryset = queryset.filter(id__in=actors)
+        if genres:
+            genres = self._params_to_ints(genres)
+            queryset = queryset.filter(genres__in=genres)
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+        if date:
+            queryset = queryset.filter(date=date)
```

Comment:
In the `get_queryset` method of `MovieViewSet`, the filter for date is likely incorrect. Ensure that the date filtering logic matches the field type and format in the `Movie` model. If `date` is not a field in `Movie`, this filter will not work as intended.